File: shared/types/customer.ts
import { Timestamp } from 'firebase-admin/firestore';
import {
  DayOfWeek,
  GenderRequirement,
  GeoLocation,
  IrregularPattern,
  PersonName,
  ServiceSlot,
} from './common';

/** 利用者 */
export interface Customer {
  id: string;
  name: PersonName;
  address: string;
  location: GeoLocation;
  ng_staff_ids: string[];
  preferred_staff_ids: string[];
  weekly_services: Partial<Record<DayOfWeek, ServiceSlot[]>>;
  household_id?: string;
  irregular_patterns?: IrregularPattern[];
  service_manager: string;
  gender_requirement?: GenderRequirement;
  kaiso_id?: string;
  karakara_id?: string;
  cura_id?: string;
  aozora_id?: string;
  phone_number?: string;
  home_care_office?: string;
  consultation_support_office?: string;
  care_manager_name?: string;
  support_specialist_name?: string;
  notes?: string;
  created_at: Timestamp;
  updated_at: Timestamp;
}

File: optimizer/src/optimizer/models/customer.py
"""利用者モデル — shared/types/customer.ts に対応"""

from pydantic import BaseModel

from .common import DayOfWeek, GeoLocation, IrregularPattern, ServiceSlot


class Customer(BaseModel):
    id: str
    family_name: str
    given_name: str
    address: str
    location: GeoLocation
    ng_staff_ids: list[str] = []
    preferred_staff_ids: list[str] = []
    weekly_services: dict[DayOfWeek, list[ServiceSlot]] = {}
    household_id: str | None = None
    irregular_patterns: list[IrregularPattern] = []
    service_manager: str = ""
    notes: str | None = None
